test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestFeatureService.java

```diff
@@ -62,9 +70,31 @@ public static boolean hasFeatureMetadata() {
 
     @Override
     public boolean clusterHasFeature(String featureId) {
-        if (hasFeatureMetadata()
-            && MetadataHolder.FEATURE_NAMES.contains(featureId) == false
-            && knownHistoricalFeatureNames.contains(featureId) == false) {
+        if (allSupportedFeatures.contains(featureId)) {
+            return true;
+        }
+        if (MetadataHolder.FEATURE_NAMES.contains(featureId) || knownHistoricalFeatureNames.contains(featureId)) {
+            return false; // feature known but not present
+        }
+
+        // check synthetic version features (used to migrate from version checks to feature checks)
+        Matcher matcher = VERSION_FEATURE_PATTERN.matcher(featureId);
+        if (matcher.matches()) {
+            Version extractedVersion = Version.fromString(matcher.group(1));
+            // FIXME consolidate with FeatureData
+            if (Version.V_8_14_0.before(extractedVersion)) {
```

Comment:
Note: This is currently not in line with the check for historical features. Attempting to define a historical feature for any version > 8.12.0 will be rejected. IMHO I think we absolutely want this to be the current version (8.14.0) to make progress here and remove the version based skipping as soon as possible before more tests are introduced based on versions rather features.

Reply:
Could this have a more detailed comment? I suspect this may be around for a while, other teams need to know what they need to do should this be hit

Reply:
Thanks, almost missed this 👍 Done